File: src/main/java/no/skatteetaten/aurora/databasehotel/service/Integration.java
package no.skatteetaten.aurora.databasehotel.service;

import java.time.Duration;

import no.skatteetaten.aurora.databasehotel.domain.DatabaseSchema;

public interface Integration {

    void onSchemaCreated(DatabaseSchema databaseSchema);

    void onSchemaDeleted(DatabaseSchema databaseSchema, Duration cooldownDuration);

    void onSchemaUpdated(DatabaseSchema databaseSchema);
}

File: src/main/kotlin/no/skatteetaten/aurora/databasehotel/service/Integration.kt
package no.skatteetaten.aurora.databasehotel.service

import java.time.Duration

import no.skatteetaten.aurora.databasehotel.domain.DatabaseSchema

interface Integration {

    fun onSchemaCreated(databaseSchema: DatabaseSchema)

    fun onSchemaDeleted(databaseSchema: DatabaseSchema, cooldownDuration: Duration)

    fun onSchemaUpdated(databaseSchema: DatabaseSchema)

    fun onSchemaReactivated(schema: DatabaseSchema)
}
